cmd/metal-api/internal/metal/network.go

```diff
@@ -303,9 +303,12 @@ func (nics Nics) FilterByHostname(hostname string) (res Nics) {
 	return res
 }
 
+// NicMap maps nic names to the corresponding nics
+type NicMap map[string]*Nic
```

Comment:
This is a useful refactoring, which could be done in a separate PR which will then reduce this PR